cli/compose/schema/data/config_schema_v3.1.json

```diff
@@ -167,7 +167,19 @@
         "ports": {
           "type": "array",
           "items": {
-            "type": ["string", "number"],
+            "oneOf": [
+              {"type": "number"},
+              {"type": "string"},
+              {
+                "type": "object",
+                "properties": {
+                  "mode": {"type": "string"},
+                  "target": {"type": "integer"},
+                  "published": {"type": "integer"},
+                  "protocol": {"type": "string"}
+                }
```

Comment:
@dnephin The `Name` is not used in the swarmkit as the moment. I added the `name` to the field. It is not actively but we could add the usage in the future.